XenAdmin/Dialogs/NetworkingPropertiesPage.cs

```diff
@@ -165,6 +167,23 @@ public void RefreshButtons()
                 haEnabledRubric.Enabled =
                     true;
             }
+
+            var pif = (PIF) Tag;
+            var existingCluster = network != null ? network.Connection.Cache.Clusters.FirstOrDefault() : null;
+
+            if (pif != null && existingCluster != null && existingCluster.network.opaque_ref == network.opaque_ref)
+            {
+                Host host = network.Connection.Resolve(pif.host);
+                    
+                var clusteringEnabled = network.Connection.Cache.Cluster_hosts.Any(cluster =>
+                    cluster.host.opaque_ref == host.opaque_ref && cluster.enabled);
+                    
+                if (clusteringEnabled && host.enabled)
+                {
+                    DisableControls("DisableForClustering",

```

Comment:
Having hard coded strings like this in the code is not a good idea, because if one of these instances changes, things wil break without warning. I would remove all these tags and either use directly the control names, or preferably, disable the parent control on which all of them are residing.